### PR TITLE
CASSANDRA-16532 Fix flaky testSkipScrubCorruptedCounterRowWithTool

### PR DESCRIPTION
**Pull request by bereng**

### CODE REVIEW

**Comment by bereng** [comment]
CI [j11](https://app.circleci.com/pipelines/github/bereng/cassandra/236/workflows/5e69db7e-7a70-4466-9669-4b8952957d28)
CI [j8](https://app.circleci.com/pipelines/github/bereng/cassandra/236/workflows/9dedf210-1ea4-4e15-88a0-2ffd63a090f3)

File: test/unit/org/apache/cassandra/db/ScrubTest.java [hunk 1]
```diff
@@ -838,42 +853,117 @@ public void testNoCheckScrubMultiRowWithTool()
...
+        assertOrderedAll(cfs, 1);
+
+        ToolResult tool = ToolRunner.invokeClass(StandaloneScrubber.class, "-e", "off", ksName, CF3);
         Assertions.assertThat(tool.getStdout()).contains("Pre-scrub sstables snapshotted into");
         Assertions.assertThat(tool.getStdout()).contains("1 rows in new sstable and 0 empty");
         tool.assertOnCleanExit();
         assertOrderedAll(cfs, 1);
     }
+
+    private void toolTestingSetup(String ksName)
```
Comment (adelapena) [inline test/unit/org/apache/cassandra/db/ScrubTest.java]: We could create the keyspace name here, and possibly also disable autocompaction, so we reduce duplication in the callers:
```java
private String toolTestingSetup()
{
    System.setProperty(org.apache.cassandra.tools.Util.ALLOW_TOOL_REINIT_FOR_TEST, "true"); // Necessary for testing
    CompactionManager.instance.disableAutoCompaction();

    String ksName = Thread.currentThread().getStackTrace()[2].getMethodName();
    createKeyspace(ksName,
                   KeyspaceParams.simple(1),
                   standardCFMD(ksName, CF),
                   standardCFMD(ksName, CF2),
                   standardCFMD(ksName, CF3),
                   standardCFMD(ksName, CF4),
                   counterCFMD(ksName, COUNTER_CF));

    return ksName;
}
```

File: test/unit/org/apache/cassandra/db/ScrubTest.java [hunk 2]
```diff
@@ -747,33 +746,19 @@ public void testFilterOutDuplicates() throws Exception
...
-
     @Test
     public void testScrubOneRowWithTool()
     {
-        CompactionManager.instance.disableAutoCompaction();
-        Keyspace keyspace = Keyspace.open(KEYSPACE2);
+        String ksName = toolTestingSetup();
+        Keyspace keyspace = Keyspace.open(ksName);
         ColumnFamilyStore cfs = keyspace.getColumnFamilyStore(CF);
         cfs.clearUnsafe();
```
Comment (adelapena) [inline test/unit/org/apache/cassandra/db/ScrubTest.java]: Do we still need the calls to `clearUnsafe` in the tool tests now that they are using a dedicated keyspace?

Also, for the other tests not using the standalone scrubber and depending on `defineSchema` perhaps we could place the call to `clearUnsafe` in an `@After` method:
```
@After
public void clearColumnFamilyStores()
{
    Keyspace.open(KEYSPACE).getColumnFamilyStores().forEach(ColumnFamilyStore::clearUnsafe);
}
```

File: test/unit/org/apache/cassandra/db/ScrubTest.java [hunk 3]
```diff
@@ -838,42 +839,106 @@ public void testNoCheckScrubMultiRowWithTool()
...
         Assertions.assertThat(tool.getStdout()).contains("Pre-scrub sstables snapshotted into");
         Assertions.assertThat(tool.getStdout()).contains("1 rows in new sstable and 0 empty");
         tool.assertOnCleanExit();
         assertOrderedAll(cfs, 1);
     }
+
+    private String toolTestingSetup()
+    {
+        System.setProperty(org.apache.cassandra.tools.Util.ALLOW_TOOL_REINIT_FOR_TEST, "true"); // Necessary for testing
+        CompactionManager.instance.disableAutoCompaction();
```
Comment (adelapena) [inline test/unit/org/apache/cassandra/db/ScrubTest.java]: I think every single test is disabling autocompaction. We could put this call in `defineSchema()` and remove the other calls.

Comment (bereng) [reply test/unit/org/apache/cassandra/db/ScrubTest.java]: Mmmm I see tests that don't such as `testScrubColumnValidation()`

File: test/unit/org/apache/cassandra/db/ScrubTest.java (hunk 2 above)
Comment (bereng) [reply test/unit/org/apache/cassandra/db/ScrubTest.java]: Some don't call clearUnsafe(), some call it mid-method, etc I remove the calls from the tool based tests, left the rest alone and triggered a run in the multiplexer. Hope it's ok.

Comment (bereng) [reply test/unit/org/apache/cassandra/db/ScrubTest.java]: The multiplexer run looks ok.

**Comment by bereng** [comment]
- CI [j11](https://app.circleci.com/pipelines/github/bereng/cassandra/241/workflows/fa465882-074e-4a9f-b55e-b18b5774c811/jobs/2468)
- CI [j8](https://app.circleci.com/pipelines/github/bereng/cassandra/241/workflows/81cbc56d-384c-435d-813e-2adf50d33b77/jobs/2471)

**Pull request closed without merging**